assembl/static2/js/app/components/common/tags/tags.jsx

```diff
@@ -0,0 +1,180 @@
+import { WithContext as ReactTags } from 'react-tag-input';
+/* eslint-enable */
+import { compose, graphql } from 'react-apollo';
+import { connect } from 'react-redux';
+import { I18n } from 'react-redux-i18n';
+// Component imports
+import AddkeywordIcon from '../icons/addkeywordIcon/addkeywordIcon';
+import CrossIcon from '../icons/crossIcon/crossIcon';
+import manageErrorAndLoading from '../manageErrorAndLoading';
+// Helper imports
+import { displayAlert } from '../../../utils/utilityManager';
+import { formatedTagList } from '../../../utils/globalFunctions';
+// GraphQL imports
+import addTagMutation from '../../../graphql/mutations/addTag.graphql';
+import removeTagMutation from '../../../graphql/mutations/removeTag.graphql';
+
+import { updateTags } from '../../../actions/tagActions';
+
+export type TagProps = {
+  /** Tag ID */
+  id: string,
+  /** Tag value */
+  text: string
+};
+
+export type Props = {
+  /** Flag that checks whether the tag is already added and displayed */
+  alreadyAdded?: string,
+  /** Flag that checks whether we have to display the admin mode */
+  isAdmin?: boolean,
+  /** List of existing tags in the overall discussion fetched and updated from the general store */
+  existingTags: Array<TagProps>,
+  initialExistingTags: Array<TagProps>,
+  /** Post ID */
+  postId: string,
+  /** List of tags related to the current post */
+  tagsList: Array<TagProps>,
+  /** Graphql mutation function called to add a new tag */
+  addTag: Function,
+  /** Graphql mutation function called to remove an existing tag */
+  removeTag: Function,
+  /** Tag list update callback: is call when a tag is added or deleted */
+  onTagListUpdateCallback: (Array<TagProps>) => void,
+  updateTags: Function
+};
+
+type State = {
+  /** List of tags related to the current post */
+  tags: Array<TagProps>
+};
+
+const KeyCodes = {
+  comma: 188,
+  enter: 13
+};
+
+const delimiters = [KeyCodes.comma, KeyCodes.enter];
+
+const RemoveComponent = () => (
+  <span className="icon-delete">
+    <CrossIcon />
+  </span>
+);
+
+const AddComponent = () => (
+  <span className="icon-add">
+    <AddkeywordIcon />
+  </span>
+);
+
+export class DumbTags extends Component<Props, State> {
+  static defaultProps = {
+    isAdmin: false,
+    alreadyAdded: 'Already added'
+  };
+
+  state = {
+    tags: [...this.props.tagsList]
+  };
+
+  handleDelete = (i: number) => {
+    const { tags } = this.state;
+    const { postId, removeTag, onTagListUpdateCallback } = this.props;
+    const selectedTag = tags[i];
+    const variables = {
+      taggableId: postId,
+      id: selectedTag.id
+    };
+    removeTag({ variables: variables })
+      .then(() => {
+        displayAlert('success', I18n.t('harvesting.tags.removeTagSuccessMsg', { tag: selectedTag.text }));
+        this.setState({
+          tags: tags.filter((tag, index) => index !== i)
+        });
+        onTagListUpdateCallback(this.state.tags);
```

Comment:
This is wrong, you're not sure this.state.tags is properly updated here, you need to pass a second argument to setState like this:
```
        this.setState({
          tags: tags.filter((tag, index) => index !== i)
        },
        () => onTagListUpdateCallback(this.state.tags));
```

Reply:
true that !